### PR TITLE
Adding google sso into relevant places on about.sourcegraph.com

### PR DESCRIPTION
**Pull request by chenkc805**
Goal: To add the ability for users to login with Google SSO, alongside places where we allow Gitlab and Github SSO login/signup on about.

### CODE REVIEW

**Comment by chenkc805** [comment]
@rrhyne can I get your help updating the ExternalsAuth.tsx file with the Google Icon? I couldn't figure out how to do that

File: src/components/AuthenticateModalContent.tsx [hunk 1]
```diff
@@ -25,6 +25,13 @@ export const AuthenticateModalContent: FunctionComponent<Props> = ({ source }) =
             className="mt-4"
```
Comment (dominiccooney) [inline src/components/AuthenticateModalContent.tsx:25]: If you're making more changes here, we could change `Gitlab` to `GitLab` on the label right ^^^ there

File: src/components/cta/ExternalsAuth.tsx [hunk 2]
```diff
@@ -51,13 +51,44 @@ const GitlabColorIcon: React.FunctionComponent<React.PropsWithChildren<{ classNa
...
 )
 
+const GoogleColorIcon: React.FunctionComponent<React.PropsWithChildren<{ className?: string }>> = ({ className }) => (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      width="46"
+      height="46"
+      viewBox="0 0 46 46"
+    >
```
Comment (dominiccooney) [inline src/components/cta/ExternalsAuth.tsx:61]: You need to pass the class name through here, to fix the warning and so your `pr-4` class where this is used below lands on a DOM element.

```suggestion
      className={className}
    >
```

File: src/components/cta/ExternalsAuth.tsx [hunk 3]
```diff
@@ -102,34 +133,63 @@ export const ExternalsAuth: React.FunctionComponent<ExternalsAuthProps> = ({
         })
     }
 
-    return authProvider === 'github' ? (
-        <Link
-            href="https://sourcegraph.com/.auth/github/login?pc=https%3A%2F%2Fgithub.com%2F%3A%3Ae917b2b7fa9040e1edd4&redirect=/get-cody"
-            className={classNames(
-                `btn hover:sg-bg-hover-external-auth-button flex items-center px-4 hover:text-black md:h-12 md:px-6 md:text-lg
+    let link: React.ReactElement;
+    link =  <Link href="" />;
```
Comment (dominiccooney) [inline src/components/cta/ExternalsAuth.tsx:137]: Nit: We could do this more succinctly, and only one space after =
```suggestion
    let link: React.ReactElement = <Link href="" />;
```

File: src/components/cta/ExternalsAuth.tsx [hunk 4]
```diff
@@ -102,34 +133,63 @@ export const ExternalsAuth: React.FunctionComponent<ExternalsAuthProps> = ({
...
+                onClick={handleOnClick}
+                id="githubButton"
+            >
+                <GithubIcon className="pr-2" />
+                {label}
+            </Link>
+        );
+    }
+
+    if (authProvider === 'google') {
```
Comment (dominiccooney) [inline src/components/cta/ExternalsAuth.tsx:157]:
```suggestion
    } else if (authProvider === 'google') {
```

File: src/components/cta/ExternalsAuth.tsx [hunk 5]
```diff
@@ -102,34 +133,63 @@ export const ExternalsAuth: React.FunctionComponent<ExternalsAuthProps> = ({
...
+                onClick={handleOnClick}
+                id="googleButton"
+            >   
+                <GoogleColorIcon className="pr-4" />
+                <span className="pr-2">{label}</span>
+            </Link>
+        );
+    }
+
+    if (authProvider === 'gitlab') {
```
Comment (dominiccooney) [inline src/components/cta/ExternalsAuth.tsx:175]:
```suggestion
    } else if (authProvider === 'gitlab') {
```

Comment (dominiccooney) [inline src/components/cta/ExternalsAuth.tsx:175]: Nit: Other places you had these in alphabetical order, but here Google is before GitLab. Consider keeping them in a consistent order.

File: src/lib/utils.ts [hunk 6]
```diff
@@ -99,14 +99,19 @@ interface ButtonTrackerData {
...
         return {
             buttonId: 'githubButton',
             conversionId: 'IE36JRZchpg6WWkoBm1cNN',
         }
     }
-
+    else if (provider === 'google') {
+        return {
+            buttonId: 'googleButton',
+            conversionId: 'someID',
```
Comment (dominiccooney) [inline src/lib/utils.ts:112]: ???

**Review by dominiccooney — APPROVED** [review]
Looks good.

I believe you need to pass `className` through to the SVG element to get that `.pr-4` on it, so take a look after you do that—you will get rightpadding where there was none before and I notice the GitHub icon has less.

Guess we need a real conversion ID instead of "someID"?

Other feedback is nits, feel free to ignore unless you're editing around there and it is convenient to tidy up.

**Comment by chenkc805** [comment]
Rebased to main and also made changes based off of @dominiccooney 's feedback, but still failing tests. This is unfortunately where my software engineering experience ends. Eventually I'll need @thenamankumar 's help getting this over the finish line

**Comment by rrhyne** [comment]
> Rebased to main and also made changes based off of @dominiccooney 's feedback, but still failing tests. This is unfortunately where my software engineering experience ends. Eventually I'll need @thenamankumar 's help getting this over the finish line

Thanks for moving this forward Kevin!

**Comment by rrhyne** [comment]
Made some updates but:

* when I test with GitLab I get "Misconfigured gitlab auth provider.". 
* When I test with GitHub I am redirected to /get-cody
* When I test with Google I am sent directly to the /search page

**Comment by rrhyne** [comment]
Additionally, the GitHub button is much smaller than the other icons and the text is off center to the icons.

**Pull request closed without merging**